go-controller/pkg/ovn/ovn.go

```diff
@@ -131,6 +135,14 @@ func (oc *DefaultNetworkController) ensurePod(oldPod, pod *corev1.Pod, addPort b
 		return oc.ensureRemotePodIP(oldPod, pod, addPort)
 	}
 
+	// If an external gateway pod is in terminating or not ready state then remove the
+	// routes for the external gateway pod
+	if util.PodTerminating(pod) || util.PodNotReady(pod) {
```

Comment:
and if it came back into ready state I guess it would hit line L185 and go onto to add things?